client/src/extension.ts

```diff
@@ -194,6 +194,18 @@ export function activate(context: ExtensionContext) {
     customCommands.switchImplIntf(client);
   });
 
+  commands.registerCommand("rescript-vscode.restartLanguageServer", () => {
+    client.stop().then(() => {
+      client = new LanguageClient(
+        "ReScriptLSP",
+        "ReScript Language Server",
+        serverOptions,
+        clientOptions
+      );

```

Comment:
What do you think about making this a function `createLanguageClient()` or similar, and re-use at the initial creation of the language client too? Just so they don't accidentally diverge.

Reply:
@zth 
can the `createLanguageClient` function accept `serverOptions,clientOptions` parameters and return the client?

I have also noticed that there is a notification handler attached to the client `rescript/compilationFinished`. 

Would it be a good idea to do this in the `createLanguageClient` function by accepting the notification handler as a callback?
or
Is it better to do this after `createLanguageClient`?

Reply:
Not sure about @zth's intention, but I would just put everything into a closure and reuse it.

```js
  let createLanguageClient = () => {
    return new LanguageClient(
      "ReScriptLSP",
      "ReScript Language Server",
      serverOptions,
      clientOptions
    );
  }
```

and then 

```js
client = createLanguageClient();
```

at least as long as you do not expect that some of the options need to be changed.

Reply:
Hi @zth I have created the `createLanguageClient` and used it.

Reply:
@a-c-sreedhar-reddy sorry, missed this! The last piece missing, I think, is that you'll need to re-instantiate the listener for the "compilation completed" event. You can test whether this is needed by using the restart command, and then after the command has restarted the client:
1. Start Dead code analysis mode
2. Make some change that fixes (or introduces) a new error that affects the dead code analysis (add a single unused let binding for instance)
3. Make sure that error is picked up and reported in the editor

The dead code analysis mode relies on the compilation completed event, so we need to make sure that's still subscribed to, even after restarts.

Reply:
@zth The dead code analysis was not getting triggered. as we stopped the old client and created a new one.

Made the changes to run deadcode analysis on `compilationFinished` event.